### ISSUE TITLE
Ranksystem with Docker

### ISSUE THREAD

**Issue by jusito** [Newcomer1989/TSN-Ranksystem#541]
Hello,

Iam using docker with php 5.6.37 & 7.2.9 and your tag 1.2.10 & 1.2.9 on linux.
The database is mariadb 10.3 user root.
Curl, zip, pdo are installed, ugo=rwx on all files for testing.

After step 2 with 1.2.10 the attached error message appears.
![1 2 10](https://user-images.githubusercontent.com/17811819/44939288-551c3380-ad84-11e8-9ff5-77f8aeb6b79c.png)
> Database error: 21S01 Array ( [0] => 21S01 [1] => 1136 [2] => Column count doesn't match value count at row 1)
But the database looks like it is created

Then I tested 1.2.9 and got to the "webinterface", the first login failed with the following message.
![ranksystem 1 2 9](https://user-images.githubusercontent.com/17811819/44939365-d96eb680-ad84-11e8-8197-fff0683c59ee.png)
> Warning: Cannot modify header information - headers already sent by (output started at /var/www/html/ranksystem/other/config.php:1) in /var/www/html/ranksystem/webinterface/index.php on line 91

Here is my repository:
https://github.com/jusito/docker-teamspeak-ranksystem/tree/develop

**Comment by Newcomer1989** [comment]
There is already a fix for that https://github.com/Newcomer1989/TSN-Ranksystem/commit/7e1c24a1efb084b6a7596ba61970c5c00b51cec6

**Comment by jusito** [comment]
Can confirm this, used:
https://codeload.github.com/Newcomer1989/TSN-Ranksystem/tar.gz/7e1c24a1efb084b6a7596ba61970c5c00b51cec6




Unfortunately, the second error also prevents me from using the system.
Maybe its not related to the installation like the first, do you want another issue opened for this?

php7, ssh2 activated.
webinterfacte/index.php after login, so first time is visible and after insertion of credentials & login it can't be opened anymore.
Since it was mentioned in one of the last issues I deleted the cookies, loading index.php first time, tried to login, failed.

stats.php, core.php, admin_addtime.php, admin_removetime.php, bot.php
![stats php7](https://user-images.githubusercontent.com/17811819/44945310-a3621e80-ade6-11e8-8028-34c821231358.png)

![core php7](https://user-images.githubusercontent.com/17811819/44945311-a65d0f00-ade6-11e8-827e-202f6e206d9a.png)

![bot php7](https://user-images.githubusercontent.com/17811819/44945332-52065f00-ade7-11e8-9e1b-18e0ea8ae872.png)

![admin_addtime php7](https://user-images.githubusercontent.com/17811819/44945333-57fc4000-ade7-11e8-9db8-e76c11d8a9f4.png)

![admin removetime php7](https://user-images.githubusercontent.com/17811819/44945336-5af73080-ade7-11e8-8623-455c4d6d6733.png)


ts.php, db.php, other.php, msg.php, addon_assign_groups.php:
Can be used, at least as far as I've tried.

**Comment by jusito** [comment]
After long research, I experienced extreme memory corruption on the server where the error occurs, I will test this again if iam sure the hardware is ok.

**Comment by Newcomer1989** [comment]
Try it again with version 1.2.11, which will be released in the next view days. There is one constellation, where the webserver could already send a session header. This will be supported with version 1.2.11.

I didn't understand, why you are using docker for this!? A normal user needs to install once a webserver and a database. I can't believe it will make sense to use docker instead in this case.
IMHO docker is in interest for hosters, which are working in the most case commercial. I don't want to support commercial hoster more than I does it already under the GNU license! The Ranksystem is free to use and should it be. No one should pay for it. I think docker will exactly bring us this!

**Comment by jusito** [comment]
I will try this, thanks a lot.

My reason for docker:
I like docker because the host is clean, you can have multiple instances of the same software without any problem, every images which works is working on your machine and every webservice is potential risk.

But the best, as we moved our dedicated we just had a script which grabs volumes(the persistend data) + the run command for each container + rsync to the new one, Rust, teamspeak, minecraft, own services aso were migrated in less than 60 minutes without issues.

2) I can work on a project, put all my knowledge in an git repository and if I need it 1-2 years ago its working with 1-2 commands - which are hopefully in the readme.

3) If there is a docker container for the ranksystem, you need 2 lines of code to set it up and know its all done, its like a install script working on every os or if done good, even working on every cpu - arm amd64. Less frustrating less stress for others and for you, for example to achieve stg. like this: https://github.com/kylemanna/docker-openvpn I would need ages.

**Comment by Newcomer1989** [comment]
The important for first:
The release 1.2.11 is out. So I hope, all of the above mentioned problems are solved. If not, please open a new issue.

Thank you for sharing your point of arguments with us.
I am still thinking, for a web application, which needs a webserver and a database, what is the absolute default on most server systems, there is a less overhead without docker.
a) First, you need to install docker
b) There would be a problem with port overlapping, when each application has his own webserver (due docker image).
c) You would have an overhead on services, when each application has his own webserver, database...
d) Without extra services you need to implement it in existing ones. There you have 1 million possibilities, how the environment looks like. That woulds make it such as hard to get these all with the docker image -> impossible. On a small scale, we already have this problem with the current semi-automatic installation process. And believe me, I know what I am talking about ;)

Docker is a good way for devs, which needs to do testing in exactly the same environment.
OR hosting companies, which needs to separate the customer from each and needs a high automatism.
And the last one is exactly what I don't want to support. :(

**Comment by jusito** [comment]
The problems above are solved!

Maybe I have time for deeper testing.

In short:
I can fully understand you  in case of hosting services and I don't want to say anything against it.

a yes
c) would be bad practice
b, d) Docker inside and outside are variable, a optimal image provides only one service and allows other containers to be used with this service

But sometimes you still want completely separated services, because overlapping could be a problem in case of security, ofc. it depends on your need and I don't think docker is good in every way.

**Comment by Newcomer1989** [comment]
Thx for the response.

And many thx for the exchange of ideas.
